inference-engine/tests/functional/plugin/cpu/single_layer_tests/eltwise.cpp

```diff
@@ -0,0 +1,259 @@
+        for (int i = 0; i < formats.size(); i++) {
+            if (formats[i] == nChw16c)
+                formats[i] = nChw8c;
+            if (formats[i] == nCdhw16c)
+                formats[i] = nCdhw8c;
+        }
+    };
+
+    if (!with_cpu_x86_avx512f()) {
+        for (auto& param : paramsVector) {
+            adjustBlockedFormatByIsa(std::get<0>(param));
+            adjustBlockedFormatByIsa(std::get<1>(param));
+        }
+    }
+
+    return paramsVector;
+}
+
+std::vector<std::vector<std::vector<size_t>>> inShapes_4D = {
+        {{2, 4, 4, 1}},
+        {{2, 17, 5, 4}},
+        {{2, 17, 5, 4}, {1, 17, 1, 1}},
+        {{2, 17, 5, 1}, {1, 17, 1, 4}},
+};
+
+std::vector<CPUSpecificParams> cpuParams_4D = {
+        CPUSpecificParams({nChw16c, nChw16c}, {nChw16c}, {}, {}),
+        CPUSpecificParams({nhwc, nhwc}, {nhwc}, {}, {}),
+        CPUSpecificParams({nchw, nchw}, {nchw}, {}, {})
+};
+
+const auto params_4D_FP32 = ::testing::Combine(
+        ::testing::Combine(
+            ::testing::ValuesIn(inShapes_4D),
+            ::testing::ValuesIn(eltwiseOpTypes),
+            ::testing::ValuesIn(secondaryInputTypes),
+            ::testing::ValuesIn(opTypes),
+            ::testing::Values(InferenceEngine::Precision::FP32),
+            ::testing::Values(CommonTestUtils::DEVICE_CPU),
+            ::testing::Values(additional_config)),
+        ::testing::ValuesIn(filterCPUSpecificParams(cpuParams_4D)));
+
+INSTANTIATE_TEST_CASE_P(CompareWithRefs_4D_FP32, EltwiseLayerCPUTest, params_4D_FP32, EltwiseLayerCPUTest::getTestCaseName);
+
+std::vector<std::vector<std::vector<size_t>>> inShapes_5D = {
+        {{2, 4, 3, 4, 1}},
+        {{2, 17, 7, 5, 4}},
+        {{2, 17, 6, 5, 4}, {1, 17, 6, 1, 1}},
+        {{2, 17, 6, 5, 1}, {1, 17, 1, 1, 4}},
+};
+
+std::vector<CPUSpecificParams> cpuParams_5D = {
+        CPUSpecificParams({nCdhw16c, nCdhw16c}, {nCdhw16c}, {}, {}),
+        CPUSpecificParams({ndhwc, ndhwc}, {ndhwc}, {}, {}),
+        CPUSpecificParams({ncdhw, ncdhw}, {ncdhw}, {}, {})
+};
+
+const auto params_5D_FP32 = ::testing::Combine(
+        ::testing::Combine(
+            ::testing::ValuesIn(inShapes_5D),
+            ::testing::ValuesIn(eltwiseOpTypes),
+            ::testing::ValuesIn(secondaryInputTypes),
+            ::testing::ValuesIn(opTypes),
+            ::testing::Values(InferenceEngine::Precision::FP32),
+            ::testing::Values(CommonTestUtils::DEVICE_CPU),
+            ::testing::Values(additional_config)),
+        ::testing::ValuesIn(filterCPUSpecificParams(cpuParams_5D)));
+
+INSTANTIATE_TEST_CASE_P(CompareWithRefs_5D_FP32, EltwiseLayerCPUTest, params_5D_FP32, EltwiseLayerCPUTest::getTestCaseName);
+
+std::vector<std::vector<std::vector<size_t>>> inShapes_4D_Mixed = {
+        {{2, 17, 31, 3}, {2, 1, 31, 3}},
+        {{2, 17, 5, 1}, {2, 1, 1, 4}},
+};
+
+std::vector<CPUSpecificParams> cpuParams_4D_Mixed = {
+        CPUSpecificParams({nChw16c, nchw}, {nChw16c}, {}, {}),
```

Comment:
Can you add more cases? like
`{nchw, nChw16c}, {nChw16c}`
`{nChw16c, nchw}, {nchw}`
`{nchw, nChw16c}, {nchw}`

Reply:
Case `{nchw, nChw16c}, {nChw16c}` - added
Cases `{nChw16c, nchw}, {nchw}` and `{nchw, nChw16c}, {nchw}` are not suported by the impl.

Reply:
ok, thanks!